step-grid-parent/step-grid-api/src/main/java/step/grid/tokenpool/TokenPool.java

```diff
@@ -213,6 +229,7 @@ protected void callListener(F object, Consumer<F> listener) {
 
 	private void removeToken(Token<F> token) {
 		tokens.remove(token.getObject().getID());
+		tokenRegistrationCallbacks.forEach(cb -> cb.afterUnregistering(List.of(token.object)));

```

Comment:
I agree in principle, but then to be consistent this would have to be done everywhere where control is "delegated", and it would be try/catches all over the place. The implicit assumption is that if a method does not declare exceptions, then it is not expected to throw any (at least not "knowingly"). We could make it absolutely clear instead in the interface (javadoc) that implementations must take care not to throw any exceptions.

Reply:
@cl-exense I agree if there's no need to guaranty that notifyWaitingPretendersWithoutMatchInTokenList() is always called after the token is removed from the map